src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/CompareSymbolsCorrectlyAnalyzer.cs

```diff
@@ -31,7 +32,27 @@ public class CompareSymbolsCorrectlyAnalyzer : DiagnosticAnalyzer
             description: s_localizableDescription,
             customTags: WellKnownDiagnosticTags.Telemetry);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+        public static readonly DiagnosticDescriptor GetHashCodeRule = new DiagnosticDescriptor(
```

Comment:
I have changed only the description for this descriptor, let me know if you would like a different message too.

Reply:
I think description change is fine for this

Reply:
Oh, I see we have the same rule ID. FWIW I don't think it hurts to separate these out into different ID. It provides more clarity even if the message is the same.

Reply:
It's never really clear when we want to split the IDs or just use different descriptors. @mavasani could you let me know what's your preference here?

Reply:
We use the following rule of thumb:
1. If the only difference is in diagnostic message, use multiple descriptors with same ID but different message.
2. If we expect user to want to bulk configure and/or suppress issues from each bucket separately, split into multiple IDs.

Reply:
@ryzngard Based on the previous comment, I don't really think we expect different bulk configuration so I would say it's better to keep the same ID but let me know if you would still prefer to have a different one.